libvast/src/table_slice.cpp

```diff
@@ -153,12 +153,20 @@ table_slice::table_slice(chunk_ptr&& chunk, enum verify verify) noexcept
       },
       [&](const auto& encoded) noexcept {
         auto& state_ptr = state(encoded, state_);
+        // We create a second chunk with an intentionally decoupled reference
+        // count here that we bind to the lifetime of the original chunk. This
+        // avoids cyclic references between the table slice and its
+        // encoding-specific state.
+        auto decoupled_chunk = chunk::make(as_bytes(*chunk_), []() noexcept {});
         auto state = std::make_unique<std::decay_t<decltype(*state_ptr)>>(
-          encoded, chunk_);
+          encoded, decoupled_chunk);
         state_ptr = state.get();
-        chunk_->add_deletion_step([state = std::move(state)]() noexcept {
-          --num_instances_;
-        });
+        chunk_->add_deletion_step(
+          [state = std::move(state), decoupled_chunk
+                                     = std::move(decoupled_chunk)]() noexcept {
+            --num_instances_;
+            static_cast<void>(decoupled_chunk);
+          });
```

Comment:
Couldn't it happen that `chunk`s reference count drops to zero and leaves `decoupled_chunk` dangling?

Reply:
I'll sketch this out on a piece of paper to think about the possible paths we can take. We definitely need to be careful here.

Reply:
You are absolutely correct. I pushed a separate commit that fixes this issue.